### PR TITLE
[feat]partially upgrade dependencies

### LINKED ISSUE #3504: [Suggestion]Better dependencies management of the repo
In the last few days, i find out that many dependencies of angular-cli and then try to get my feet wet.

And I just find @hansl #3488 also updates some of the dependencies. But I think it's less perfect.

---
### My opinions
As the repo is a host for several packages, what does the `dependencies` means for the repo itself? It will not be published.

Let's say the repo is the host.

In fact, the host is responsible for **linting**, **building** and **testing** the real-world packages. Without tests, the host theoretically ought to have `devDependencies` only.

However, in real development, developers need to run unit tests and write code with the power of Typescript and IDE. So, we have to install `dependencies` of those packages. (**Note:** e2e tests don't require that because they install dependencies independently)

Therefore, it can be concluded that the `dependencies` field in the `package.json` of the host itself, should be the combination of the  `dependencies` of those several packages. And then, the update workflow of the dependencies should be:
* If they are the dependencies of those packages, developer should update them in those packages' `package.json`
  * then, make a join of the packages' `dependencies` and make a update in host's `package.json`
  * one should not directly add dependencies into `dependencies` field of host's `package.json`
* If they are the `devDependencies`, of course they should be updated in `devDependencies` in `package.json` of repo itself.
* If there is duplication between `dependencies` and `devDependencies`, I would perfer to remove it from `devDependencies`

The currents status is, lacking a util to sync the `dependencies` from the packages and a note to warn contributor not to manually update dependencies in repo's `package.json`

### PR DESCRIPTION
**Pull request by e-cloud**
The PR upgrade some of dependencies that doesn't need big code changes, and provide a script for better dependencies management.

Fix #3504

### CODE REVIEW

**Comment by e-cloud** [comment]
I signed it!

**Comment by hansl** [comment]
Hi. Please rebase; we updated a bunch of other dependencies already. I'll review once this PR is up to date.

**Comment by e-cloud** [comment]
@hansl it's rebased now.

**Comment by hansl** [comment]
Seems like this PR is a mess now. The changes to validate_dependencies are weird and the sync-deps doesn't seem to work...

**Comment by e-cloud** [comment]
My mistake! @hansl sorry, what i want is to check the dependencies version across the packages in this repo. Perhaps i should not exit with 1, a warning it's better.

Do you think it's reasonable? What's the weird part you are talking about?

**Comment by e-cloud** [comment]
@hansl please have a review :).

**Comment by robertbaker** [comment]
@e-cloud fix conflicts please

**Comment by e-cloud** [comment]
conflicts resolved and I don't upgrade the yarn.lock

**Comment by e-cloud** [comment]
@hansl Would you give some time to review this? or just let it be? You can close this PR if not your desire.

**Comment by e-cloud** [comment]
@hansl request for review

File: scripts/publish/validate_dependencies.js [hunk 1]
```diff
@@ -28,6 +29,18 @@ const ANGULAR_PACKAGES = [
...
 
+let exitCode = 0;
+
+const depsInfo = validateDependenciesPerPackage(packages);
+
+validateRootPackageJson(depsInfo.allSourceDeps);
+
+checkDependenciesConflictAcrossPackages(depsInfo.allDeclaredDepsMap);
+
+process.exit(exitCode);
```
Comment (intellix) [inline scripts/publish/validate_dependencies.js]: Did you mean to leave this here?

**Comment by e-cloud** [comment]
will split the PR into two

**Pull request closed without merging**